src/lib.rs

```diff
@@ -68,8 +70,12 @@ impl Drop for ShmemConf {
 #[allow(clippy::new_without_default)]
 impl ShmemConf {
     /// Create a new default shmem config
-    pub fn new() -> Self {
+    pub fn new(droppable: bool) -> Self {
```

Comment:
In terms of trying to keep the builder pattern working as intended (and also to allow this pr not to be breaking (whenever we do propose this to be added to the library)), it might be best to set `droppable` to `true` by default and add a method like `non_droppable` to `ShmemConf` to toggle it off. I'm fully aware that this here is the same change we added to our vendored version. I'm also not proposing that this change should be done now (and advising that we shouldn't, mostly because then we would need to do changes in the code of our repo that uses this forked lib). Instead maybe we should leave a TODO here to fix this later.